ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -100,10 +100,28 @@ data Mempool m blk = Mempool {
       -- index of transaction hashes that have been included on the blockchain.)
       addTxs :: [GenTx blk] -> m [(GenTx blk, ApplyTxErr blk)]
 
-      -- | Get all transactions in the mempool (oldest to newest)
+      -- |
+    , syncState :: STM m ()
+
+      -- | Get all transactions in the mempool along with their associated
+      -- ticket numbers (oldest to newest).
       --
-      -- Guarantees that these transactions will be valid (when applied strictly
-      -- in order) with respect to a call to 'getLedgerState' run /in the same
-      -- transaction/.
-    , getTxs :: STM m (Seq (GenTx blk))
+      -- n.b. This function provides /no guarantee/ that the resulting
+      -- transactions will be valid. However, it is guaranteed that these
+      -- transactions will be valid if this function is called immediately
+      -- following a call to 'syncState', /within the same transaction/.
+    , getTxs :: STM m (Seq (GenTx blk, idx))
```

Comment:
Do we need a `Seq` here (same for `getTxsAfter`)? If the consumers of this API are just as happy accepting a list (e.g., they mostly call `toList`), I would avoid the extra overhead of `fromTxSeq`.

Reply:
Yeah, I think both consumers are happy with a list.